### ISSUE TITLE
Need to validate data when doing Table.update

### ISSUE THREAD

**Issue by csdev** [ryanfitz/vogels#33]
The `Table.create` method uses joi to check that all of the supplied data is valid before attempting to serialize it and write it to the database ([table.js line 116](https://github.com/ryanfitz/vogels/blob/a9d42a65389025b91e78ab30177e25f01f10bed1/lib/table.js#L116)). However, `Table.update` does not do any data validation. This can result in invalid data being saved to the db. It can also cause a crash since some of the serializer code assumes that it is working with valid data. (For example, if the user performs an update with an invalid Date, a later call to `Date.toISOString` will raise an uncaught error.)

**Comment by viniciuskneves** [comment]
@ryanfitz I've got on similar issue this week and some coworkers have faced it too in the past. I've two different suggestions that could be implemented (I could open the PR if you prefer and don't have time to it). First one would be add an option like 'updateValidation' (false as default) and it would be able validation before update. The problem that I see on this feature is that it is a little bit intrusive.
The second one would be able to return the object schema and them it would be able to validate it before performing an update whenever the developer feels it is necessary.

Would like to know your opinion. I'm adapting (not modifying the lib) the second solution over here as we've faced some problems...

**Comment by avtaniket** [comment]
@viniciuskneves Yes, this issue need to be handled, an record update is important operation as much as create, so we must validate data before update.

We should have separate mechanism to validate data on update eg. updateValidations() or validateOnUpdate(true) during schema.

**Comment by Alex0007** [comment]
validation mechanism in vogels is incomplete without this

+1 for this feature

**Comment by speedster-kiev** [comment]
+1 for feature

**Comment by andyday** [comment]
+1 for feature

**Comment by sjungwirth** [comment]
+1
I would also like to do on demand validation, before making the create request, which currently seems impossible due to table/schema being private.

**Comment by AaronHarris** [comment]
@sjungwirth I finally found you can access the table/schema by calling model.scan(), which returns an object with the table, which contains the schema object, which you can call validate. It's hacky but it seems the easiest way to access the object.

**Comment by sjungwirth** [comment]
@AaronHarris this is really helpful, thank you. While looking into what you said it looks like there is a reference to `table` on model instances as well. So it looks like either of these will work:

`myModelInstance.table.schema.validate(myModelInstance)` or `MyModel.scan().table.schema.validate(myDataOrModelInstanceToValidate)`
